backend/tests/unit/dal/Services/ProjectServiceTest.cs

```diff
@@ -579,7 +580,7 @@ public void Update()
 
             // Assert
             Assert.NotNull(result);
-            Assert.Equal(projectToUpdate, result);
+            result.Should().BeEquivalentTo(projectToUpdate, options => options.Excluding(o => o.SelectedMemberPath.Contains("Updated")));
```

Comment:
So, I was looking at these tests and wondering why they were just using Assert.Equal(object1, object2), which just compares the object references, instead of Assert.Equal(object1, object2, new DeepPropertyCompare()). It turns out that a number of these tests crash when using the DeepPropertyCompare. Also, even when the test doesn't crash it is hard to tell what is causing the equality to fail as the assert message isn't helpful and debugging large, nested entities to find the differences isn't a good solution.

I can think of two solutions:
1) stop using equals on large objects, assert on general features of those objects instead (ie, the objects id, type, or number of values in the case of an array)
2) Use the library that I found above, https://fluentassertions.com/ allows deep comparison of nested objects with the ability to exclude/include properties via configuration. This library also gracefully handles circular references. It also provides specific feedback on the exact portions of the objects that failed equality testing.

Reply:
I'm a fan of FluentAssertions.  Using Assert.Equal with a comparer was providing some of the similar benefits of FluentAssertions.

Reply:
I also like fluent assertions